tests/functional/test_install.py

```diff
@@ -1391,6 +1391,41 @@ def test_install_pep508_with_url_in_install_requires(script):
     assert "Successfully installed packaging-15.3" in str(res), str(res)
 
 
+@pytest.mark.network
```

Comment:
I think it would be better not to add more tests that depend on the network. Can this be done with local packages and `file://` urls instead?

Reply:
It doesn't look like it. Given a `setup.py` of:

```python
from setuptools import setup
kwargs = {'name': 'pkga', 'version': '1.0', 'install_requires': ['dep@file:///private/var/folders/bw/ghklm2fs6bx_x9x_f0wg5kph0000gn/T/pytest-of-andy/pytest-15/test_install_pep508_with_url_i0/workspace/scratch/dep-1.0-py2.py3-none-any.whl']}
setup(**kwargs)
```

The tests fail with:

```
Script result: python -m pip install /private/var/folders/bw/ghklm2fs6bx_x9x_f0wg5kph0000gn/T/pytest-of-andy/pytest-15/test_install_pep508_with_url_i0/workspace/scratch/pkga
  return code: 1
-- stderr: --------------------
    ERROR: Complete output from command python setup.py egg_info:
    ERROR: error in pkga setup command: 'install_requires' must be a string or list of strings containing valid project/version requirement specifiers; Invalid URL given
    ----------------------------------------
ERROR: Command "python setup.py egg_info" failed with error code 1 in /private/var/folders/bw/ghklm2fs6bx_x9x_f0wg5kph0000gn/T/pytest-of-andy/pytest-15/test_install_pep508_with_url_i0/workspace/tmp/pip-req-build-qv9tysxp/
```

Reply:
I was just making a suggestion. Can you come up with a way to make it work without hitting the network? Also, is there a bug in something causing what you tried not to work?

Reply:
> Also, is there a bug in something causing what you tried not to work?

I think the issue is that setuptools is vendoring an old version of `packaging` and the bug not accepting `file://` URLs there was fixed in `packaging==19.0` (https://github.com/pypa/packaging/commit/eb0243865079c7c2179730da93130447556ea020) 😩 

 I figured out how to make it work by skipping `setup.py` and just building wheels though 😌